STACpopulator/populator_base.py

```diff
@@ -76,6 +76,13 @@ def stac_host(self) -> str:
     def collection_id(self) -> str:
         return self._collection_id
 
+    @property
+    @abstractmethod
+    def item_properties_model(self):
+        """In derived classes, this property should be defined as a pydantic data model that derives from
+        models.STACItemProperties."""
+        pass
```

Comment:
`raise NotImplementedError` instead

Reply:
That's not needed, the body of the code is never executed. You cannot instantiate an abstract class or a derived class where the abstract method is not defined. Python raises a `TypeError` in both cases.